### ISSUE TITLE
Unwanted highlighting while on insert mode

### ISSUE THREAD

**Issue by tomasyany** [ntpeters/vim-better-whitespace#36]
Sometimes, and I'm trying to understand when and how, the plugin starts highlighting end-of-line whitespace while on insert mode.
For example, if "_" is the white space, then while on insert mode, I get a highlight of the last space on the following --> "I'm_writing_this_line_". 

The process would be (always while on insert mode):
I'm_writing --> no highlighting
I'm_writing_ --> highlights the last white space
I'm_writing_th --> no highlighting

This is pretty annoying because there is a red highlight moving everytime one presses the space bar.

I'm trying to reproduce the problem but can't figure it out what makes it happen. 
Closing and open vim solves it.

**Comment by ntpeters** [comment]
Alright, I think I've figured out how to reproduce this.  If you `:DisableWhitespace` then `:EnableWhitespace`, this seems to cause this behavior for some reason.  Are you calling these at all while editing? Or have them being called in your `.vimrc`?

I still need to look into the root cause of this, but hopefully I'll push out a fix soon-ish.

**Comment by alexbel** [comment]
I can see highlighted whitespaces in insert mode if I open quickfix window (`:copen`)

**Comment by vyp** [comment]
Hmm I was editing my vimrc today, and I got this just once when I was typing a comment (iirc I was adding some words to an already existing comment). Unfortunately I cannot seem to reproduce this now. Also calling `:DisableWhitespace` and then `:EnableWhitespace` and then editing again does not seem to reproduce this for me either. So I guess I can't be of much help rather than to just say me too..

**Comment by tomasyany** [comment]
So far I have **solve it** by exactly doing the disable/enable thing.

**Comment by dragonxlwang** [comment]
join this discussion to see if i can help to reproduce:

in .vimrc
``
" remove trailing whitespace
Plugin 'ntpeters/vim-better-whitespace'
" https://github.com/ntpeters/vim-better-whitespace/issues/40
" add following to turn on when start
autocmd VimEnter * DisableWhitespace
autocmd VimEnter * EnableWhitespace
" turn on by default for all filetypes
autocmd BufWritePre * StripWhitespace
``

In this case, when writing a new file i can always see in insert mode that the current line is being highlighted unwantedly.

**Comment by dragonxlwang** [comment]
I think I have found the solution. maybe i didn't understand the setup very correctly --- my ideal setup is highlight is enabled by default but disabled only in **insert mode**. Also I want the trailing whitespace being stripped on save.

If this is the case, then the problem:

- highlight is not shown on vim start: this is becuase ``s:EnableWhitespace()`` is not called by default. To call this, one need first set `` g:better_whitespace_enabled = 0`` as otherwise the function returned directly

To sum up, the following setup works for me.

```
" remove trailing whitespace
Plugin 'ntpeters/vim-better-whitespace'
" add following to turn on when start
let g:better_whitespace_enabled = 0
autocmd VimEnter * EnableWhitespace
" turn on by default for all filetypes
autocmd BufWritePre * StripWhitespace
```

**Comment by ntpeters** [comment]
Fixed in #42

**Comment by jdhao** [comment]
I am also using this plugin. I have found that in insert mode, space after text will be highlighted in red. But I have disable it by using

```
let g:better_whitespace_enabled = 0
```

It turns out that the plugin [python-syntax](https://github.com/hdima/python-syntax#options-used-by-the-script) will also try to highlight the trailing space.  To disable the highlighting, use 

```
let g:python_highlight_space_errors=0
```
